### DISCUSSION TITLE
Generate data with Valid Medical Conditions

### DISCUSSION THREAD

**Discussion by mobashshir005** [synthetichealth/synthea#1503] [General]
in generated data, I see condition resource type has medical conditions as full_time_employment, part_time_employment, transfered to skilled nursing facility etc. how to generate records having valid medical condition?

**Comment by jawalonoski** [comment]
Those conditions are **Social Determinants of Health (SDoH)**. Arguably, they are valid medical conditions, after all they have clinical terminology codes associated with them.

In Synthea, this data is "collected" through the PRAPARE questionnaire: https://prapare.org/the-prapare-screening-tool/

The module that generates the data is here: https://synthetichealth.github.io/module-builder/#encounter/sdoh_hrsn

The records you generate should also contain traditional diagnoses like "Diabetes" and "Hypertension" and so forth, as appropriate for the individual.

**Reply by mobashshir005** [reply]
OK got it, but it is too frequent. Can we exclude this module?

**Reply by jawalonoski** [reply]
There is no easy way to exclude this module as of September 2024, such as a command-line parameter.

Here are two options...

- You could modify the [wellness encounter](https://synthetichealth.github.io/module-builder/#wellness_encounters) module, and change the `PRAPARE` state to `Simple` instead of `CallSubmodule`. Save the new model to `./src/main/resources/modules/wellness_encounters.json`
- You could replace `./src/main/resources/modules/encounter/sdoh_hrsn.json` with the contents of an empty module:

```json
{
  "name": "Untitled",
  "remarks": [
    "A blank module"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "direct_transition": "Terminal"
    },
    "Terminal": {
      "type": "Terminal"
    }
  },
  "gmf_version": 2
}
```
